elasticdl/python/worker/worker.py

```diff
@@ -314,39 +314,32 @@ def get_model(self):
             self._distribution_strategy
             == DistributionStrategy.PARAMETER_SERVER
         ):
-            variable_future_and_id_pairs = []
-            for ps_id, stub in enumerate(self._ps_stubs):
-                if ps_id not in self._ps_client.ps_to_parameter:
-                    continue
-                # async grpc call
-                req = elasticdl_pb2.PullDenseParametersRequest()
-                req.version = self._model_versions_from_ps[ps_id]
-                var_future = stub.pull_dense_parameters.future(req)
-                variable_future_and_id_pairs.append((var_future, ps_id))
-
-            for var_future, ps_id in variable_future_and_id_pairs:
-                res = var_future.result()
-                if not res.initialized:
-                    # push variable to ps for initialization
-                    parameters = [
-                        Tensor(name, self._non_embed_vars[name].numpy(), None)
-                        for name in self._ps_client.ps_to_parameter[ps_id]
-                    ]
-                    self._ps_client.push_dense_parameters(
-                        parameters, ps_id, self._model_versions_from_ps[ps_id]
+            dense_params, uninit_ps = self._ps_client.pull_dense_parameters(
+                [i for i in range(self._ps_num)], self._model_versions_from_ps
+            )
+
+            for ps_id in uninit_ps:
```

Comment:
How about adding more comments for these two steps
1) First pull dense parameters from all the PS node, maybe one or more ps is uninitialized
2) Then we will push the dense parameter from the worker pod to these Pservers to initialize its partition of parameters.

Reply:
Good idea!

Reply:
How about make this procedure three steps:
1) pull_dense_parameter from all the ps
2) Initialize the uninit ps if there is any
3) pull_dense_parameters from all the ps again and then assign _non_embed_vars

Reply:
After the PS is initialized, we need not call `push_dense_parameters`. So, it's a waste of time to call `pull_dense_parameters` twice.

Reply:
The second pull_dense_parameters is only called when there is any uninitialize PS node. Just like initialize the singleton if the instance is null

Reply:
@brightcoder01 I will continue polish codes in the next PR.

Reply:
Yes, the code can be revised to be more clear such as:

```
dense_param, unit_ps = pull_dense_param(all_ps)

if uninit_ps:
   initialize_ps_nodes(uninit_ps)
   more_dense_param, unit_ps = pull_dense_params(uninit_ps)
   assert(len(unit_ps) == 0
   dense_param.update(more_dense_param)

assign_nonembed_var(dense_param)

```